### PR TITLE
[FLINK-19982][test] Change the log when CollectSinkFunction receives an invalid request from debug to info for future investigation

### PR DESCRIPTION
**Pull request by TsReaper**
## Brief change log

 - Change the log when CollectSinkFunction receives an invalid request from debug to info for future investigation of FLINK-19982.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/collect/CollectSinkFunction.java [hunk 1]
```diff
@@ -383,12 +383,10 @@ public void run() {
...
 					if (!version.equals(requestVersion) || requestOffset < offset) {
 						// invalid request
-						if (LOG.isDebugEnabled()) {
-							// this is normal for the 1st request after the sink (re)starts, so we print debug log
-							LOG.debug("Invalid request. Received version = " + requestVersion +
-								", offset = " + requestOffset + ", while expected version = "
-								+ version + ", offset = " + offset);
-						}
+						// this is normal for the 1st request after the sink (re)starts
+						LOG.info("Invalid request. Received version = " + requestVersion +
```
Comment (wuchong) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/collect/CollectSinkFunction.java:387]: How many these logs will there be? Will it flow the log file?

Comment (wuchong) [reply flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/collect/CollectSinkFunction.java:387]: Use `LOG.warn`?

Comment (TsReaper) [reply flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/collect/CollectSinkFunction.java:387]: It won't flow the log file. This log only happens when the sink starts or restarts.

This is a mechanism to inform the collect iterator that the sink has restarted and is an expected behavior. So an info log is enough.

**Comment by TsReaper** [comment]
@flinkbot run azure

**Review by wuchong — APPROVED** [review]
LGTM.

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/collect/CollectSinkFunction.java (hunk 1 above)
Comment (wuchong) [reply flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/collect/CollectSinkFunction.java:387]: Thanks for the explanation.

**Pull request merged**